### DISCUSSION TITLE
Question/suggestion about DOI Versioning based on Zenodo’s approach

### DISCUSSION THREAD

**Discussion by jeffpooley** [knowledgefutures/pubpub#1252] [Ideas]
This post is a question that, depending on the answer, could be a suggestion. 

One of the many things I love about PubPub is its support for versions and version histories, in Git-inspired fashion. The built-in DOI-minting is really well done too.

Here's the question: Does PubPub already support the [Zenodo approach to DOI versioning](https://help.zenodo.org/#versioning)? The basic idea is to register a single all-versions DOI and then register additional DOIs for each new specific version. The original, all-versions DOI, [in their model](https://help.zenodo.org/#versioning), always resolves to the latest version. 

Assuming that PubPub doesn't work this way—I think not, but I don't want to tempt the Crossref gods by playing around—I wonder if this is something that the team would consider? Are others interested in a DOI versioning system like this?

**Comment by gabestein** [comment]
Thanks for the idea! We actually used to do this using component DOIs, but Crossref didn't love it (both, I think, because we weren't using Crossmark and because we were hacking component DOIs to do it avoid paying $1 per version), so we stopped the practice. I think Zenodo uses Datacite, which has different fee structures.

I can definitely see us bringing it back by minting a new one on each release. I think the Crossref fee structure might have to change a bit to make that viable, though (which I think is in the works). One thing that might help us make the case both internally and to Crossref is knowing more about the use-case. Is it mainly for citability of versions, or did you have some other ideas for it in mind?

**Comment by jeffpooley** [comment]
Thanks Gabe! To answer your question, the use case I had in mind wasn’t for a new DOI per release, but instead the option to manually mint a new DOI when a Pub has reached a sufficiently different version (a subjective judgment, but akin to a point release for software). In the ideal setup, there would be (like Zenodo), a single all-versions DOI that always resolves to the latest, and then also version-specific DOIs, for citability. (Though I’m not sure how the Zenodo model of all-versions + version-specific DOIs involves, if at all,  “component DOIs”.) It’s less a high priority on my side than curiosity about whether it’s already possible somehow.

Jeff

**Reply by gabestein** [reply]
Ah, that's really interesting! We've talked about a way to label or tag different releases a la GitHub (you sort of can now, with the release notes). I can see this idea potentially being an extension of that, but perhaps a decision made at release-time rather than automatic, due mostly to the cost.

Going a tad deeper, how would you imagine folks using the concept of tagged releases in the wild? Would you want to see some updates to the Pub page to make it clear this is considered a "sufficiently different version"? Do you think folks would want to compare old and new? Or would some people, you think, continue to reference the older ones as it updated?

**Comment by jeffpooley** [comment]
I realize I never responded to this follow up on the versioning. I suppose that Github-style tagging would be a way to do this, with the option for “1.1” or the like.  Ideally you would be able to cite each “sufficiently different version”, with its own version DOI—with a parent DOI always resolving to the latest version, in the Zenodo mold.

So the idea would be to be able to cite a book or article’s version 1.2, as well as, say, it’s 1.5 version, in the way that editions work with books. The F1000 version system seems clunky but aims for something like this.

My understanding of the current “Update & Preview Deposit” in the Pub Settings menu (a really nicely done UI, by the way) is that it just updates the DOI metadata.

In the case of books, of course, it would be great to have this versioning option at the Collection level.
